cmd/snap/inhibit.go

```diff
@@ -0,0 +1,152 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"time"
+
+	"github.com/snapcore/snapd/cmd/snaplock/runinhibit"
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/progress"
+)
+
+func inhibitMessage(snapName string, hint runinhibit.Hint) string {
+	switch hint {
+	case runinhibit.HintInhibitedForRefresh:
+		return fmt.Sprintf(i18n.G("snap package %q is being refreshed, please wait"), snapName)
+	default:
+		return fmt.Sprintf(i18n.G("snap package cannot be used now: %s"), string(hint))
+	}
+}
+
+func isGraphicalSession() bool {
+	return os.Getenv("DISPLAY") != "" || os.Getenv("WAYLAND_DISPLAY") != ""
+}
+
+var hasZenityExecutable = func() bool {
+	return osutil.ExecutableExists("zenity")
+}
+
+func zenityFlow(snapName string, hint runinhibit.Hint) error {
```

Comment:
is the intention that we do not want to test the zenity flow because it will very shortly be replaced with a D-Bus implementation? as it is I don't see any tests exercising the zenity flow logic ?

Reply:
Yes, zenity will go away soon.

Reply:
not sure if I mentioned my opinion earlier, but at this stage I would much rather just merge this PR with zenity so that it can be merged and then open smaller followups which remove zenity and replace it with D-Bus etc so we can make more progress without having to re-review this same PR again